### PR TITLE
Add BUCash 1.6.0 release notes

### PR DESCRIPTION
**Pull request by sickpig**
Just a very early draft, feedback are very welcome.

### CODE REVIEW

File: doc/release-notes-bucash.md [hunk 1]
```diff
@@ -32,162 +33,364 @@ If you are upgrading from a release older than 1.1.2.0, your UTXO database will
...
+- Graphene: seeding of IBLT hash function
+- Graphene: use cheap hashes to avoid short id tx collision
+- Compact Block implementations (BIP 152)
+- Easier gitian build process (docker based)
+- Code restructure around the 3 block propagation techs we currently support (Graphene, CB, Xthin)
+- Move to C++14 and bump lib boost min ver to match
+- Update Windows build scripts
+- Simplify and update expedited code
+- Refactor Script interpreter as a "virtual machine" encapsulated by a class
+- implement xversion changeable key
```
Comment (ptschip) [inline doc/release-notes-bucash.md:66]: what about the electrum integration...aren't we waiting for that to be merged still?

File: doc/release-notes-bucash.md [hunk 2]
```diff
@@ -32,162 +33,364 @@ If you are upgrading from a release older than 1.1.2.0, your UTXO database will
...
+- Graphene could use Fast Filter instead of Bloom Filter
+- Graphene: seeding of IBLT hash function
+- Graphene: use cheap hashes to avoid short id tx collision
+- Compact Block implementations (BIP 152)
+- Easier gitian build process (docker based)
+- Code restructure around the 3 block propagation techs we currently support (Graphene, CB, Xthin)
+- Move to C++14 and bump lib boost min ver to match
+- Update Windows build scripts
+- Simplify and update expedited code
+- Refactor Script interpreter as a "virtual machine" encapsulated by a class
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: add ".  This allows the script interpreter to be used outside of bitcoind (for example, its ability to "step" allows use in a script debugger).

File: doc/release-notes-bucash.md [hunk 3]
```diff
@@ -32,162 +33,364 @@ If you are upgrading from a release older than 1.1.2.0, your UTXO database will
...
-
-- Documentation fixes
-- Make it possible to disable graphene and xthin preferential timer
-- Switch ctor on earlier in the init process if we are on the BCH chain
-- Improve Graphene decode rates by padding IBLT
-- Improve xversion QA testing
-- General improvements and fixes to the QA test framework
+- Add SCRIPT_ALLOW_SEGWIT_RECOVERY (May 15th upgrade)
+- Schnorr signature scheme (May 15th upgrade)
+- RPC enhancements
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: @Greg-Griffith can you summarize the changes in the batch-tx-rpc branch here

Comment (Greg-Griffith) [reply doc/release-notes-bucash.md]: That PR added 3 new rpc commands:
- getrawblocktransactions
  - getrawblocktransactions takes the block hash and a protocol id (optional) and returns the raw transaction data of all transactions in the block. If a protocol id is supplied then it will not return any transactions that do not have an OP_RETURN that meets the protocol id criteria. Use * as a wildcard for any id.

- getrawtransactionssince
  - getrawtransactionssince takes the blockhash, block count, and a protocol id (optional) and returns the raw transaction data for <count> blocks starting with <blockhash> and moving towards the tip.  If a protocol id is supplied then it will not return any transactions that do not have an OP_RETURN that meets the protocol id criteria. Use * as a wildcard for any id.

- gettxoutproofs
  - gettxoutproofs takes a list of transaction hashes and a blockhash. it returns a hex-encoded proof for each transaction in the given list of transaction hashes that is inside the block with hash <blockhash>..

File: doc/release-notes-bucash.md (hunk 1 above)
Comment (sickpig) [reply doc/release-notes-bucash.md:66]: I don't think we should wait for it to release 1.6. I think we need as much run time as possible for 1.6. So the earlier we release the better. 

We could add [electrs](https://github.com/dagurval/electrs) integration in a subsequent release (e.g. 1.6.0.1)

File: doc/release-notes-bucash.md (hunk 3 above)
Comment (sickpig) [reply doc/release-notes-bucash.md]: Will add these new 3 rpc calls to the list of changes. I don't think we should add also the entire description for all of them, thou.

**Comment by gandrewstone** [comment]
add electrum as an experimental feature (off by default) and some small instructions:

"make electrs" to build

bitcoin.conf:
electrum=1  # to start the electrum server
electrumexec=path/to/bin/electrs  # optional path to your electrum server binary

Example of how to make a local connection to the server:
./electron-cash -v --oneserver --server=127.0.0.1:60001:t --testnet

Known issue: your RPC password is shown in the electrs command line.

**Comment by sickpig** [comment]
> add electrum as an experimental feature (off by default) and some small instructions:

done
